src/ProtoRoute/Message.hs

```diff
@@ -41,6 +54,33 @@ fromTV (Opt txt)  = show txt
 fromTV (Rep list) = show list
 fromTV Error = "Error"
 
+checkFields :: (FieldValue, (OnlyFRule, OnlyFType)) -> Bool
+checkFields ((FText tvA), (rA, fTA)) = fTA == Sstr && (correct tvA rA)
+checkFields ((FInt tvB), (rB, fTB))  = fTB == Number && (correct tvB rB)
+checkFields ((FMsg tvC), (rC, fTC))  = fTC == Msg && (correct tvC rC)
+
+correct :: (Eq a) => TValue a -> OnlyFRule -> Bool
+correct (Req _) TReq = True
+correct (Opt _) TOpt = True
+correct (Rep _) TRep = True
+correct _ _ = False
+
+-- Not quire sure how to check for empty required fields here
+validateMessage :: Message -> MessageSchema -> Either ValidationError ()
+validateMessage (Message mn fields) (MessageSchema mns sFields) =
+    if wrongMsgNames
+    then Left (WrongName "Message names are different")
```

Comment:
This patterns occurs very often. You can write function like:

```haskell
verify :: Bool -> e -> Either e ()
```

and use `do`-notation to write these things shorter.